lang/src/org/partiql/lang/eval/time/Time.kt

```diff
@@ -120,6 +120,24 @@ data class Time(val localTime: LocalTime, val precision: Int = MAX_PRECISION_FOR
             OffsetTime.of(localTime, it)
         }
 
+    /**
+     * Returns the TIMEZONE_HOUR for the [zoneOffset] of this instance.
+     */
+    val timezoneHour
+        get() : Int? = zoneOffset?.totalSeconds?.div(SECONDS_PER_HOUR)
+
+    /**
+     * Returns the TIMEZONE_HOUR for the [zoneOffset] of this instance.
+     */
+    val timezoneMinute
+        get() : Int? = (zoneOffset?.totalSeconds?.div(SECONDS_PER_MINUTE))?.rem(SECONDS_PER_MINUTE)
+
+    /**
+     * Returns the seconds along with the fractional part of the second's value.
+     */
+    val secondsWithFractionalPart
+        get()  = BigDecimal(localTime.second + localTime.nano / NANOS_PER_SECOND)
```

Comment:
I think this is problematic (but maybe tests exist that prove me wrong). Aren't you doing integer division here so isn't that effectively zero?  I think what you want is is to scale the coefficient (`second + nano`) by the scale `9`.  But you probably should take into account precision if returning this as a `BigDecimal`.

Reply:
`NANOS_PER_SECOND` is a `double` value so it's not an integer division. 
`secondsWithFractionalPart` is already rounded to the given `precision` as this is taken care of when the `Time` instance is created.

Reply:
Don't mix float and decimal math, you will be creating weird rounding out of your control.

Reply:
Not sure what you meant here. Do you mean being explicit here ? Something like this? 
```
    val secondsWithFractionalPart
        get()  = BigDecimal(localTime.second.toDouble() + localTime.nano / NANOS_PER_SECOND)
```

Reply:
Meaning, I would not use `double` at all in any of this, keep to `BigDecimal` math.